python/pyarrow/_dataset.pyx

```diff
@@ -1046,6 +1046,21 @@ cdef class FileFragment(Fragment):
             self.partition_expression
         )
 
+    def slice(self, start, end):
+        """
+        Returns a new FileFragment object that will only read 
+        a slice of the old FileFragment defined by start (start byte)
+        and end (end byte).
+        """
+
+        cdef FileFragment new_fragment = self.format.make_fragment(
+            self.path if self.buffer is None else self.buffer,
+            self.filesystem,
+            self.partition_expression
+        )
+        new_fragment.file_fragment.set_bounds(start, end)
```

Comment:
Hmm... what happens if I call slice() several times (e.g. `fragment.slice(1, 10).slice(2, 11)`)?

Reply:
Yeah I am going to update this to subslice, where subslicing logic is implemented in Python